doc/config_rsmexplain.rst.inc

```diff
@@ -0,0 +1,66 @@
+
+background_kmeans_size *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The size of the k-means sample for background. Defaults to ``500``. A dataset is summarized with k mean samples weighted by the number of data points they each represent.
+
+custom_sections *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A list of custom, user-defined sections to be included into the final report. These are IPython notebooks (``.ipynb`` files) created by the user.  The list must contains paths to the notebook files, either absolute or relative to the configuration file. All custom notebooks have access to some :ref:`pre-defined variables <custom_notebooks>`.
+
+description *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~
+A brief description of the ``rsmexplain`` experiment that will be shown at the top of the report. The description can contain spaces and punctuation.
+
+num_features_to_display *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Number of top features that should be displayed in ``rsmexplain`` plots. Defaults to ``15``.
+
+id_column *(Optional)*
+~~~~~~~~~
+The name of the column containing the response IDs. Defaults to ``spkitemid``, i.e., if this is not specified, ``rsmexplain`` will look for a column called ``spkitemid`` in both background_data and explainable_data files. Note: id_column must be same in these two files.
+
+sample_range *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~~
+If we want to explain a specific range of responses from the ``explainable_data``, we can specify that range here. Note that the range is specified in terms of the *location* of the responses in the ``explainable_data`` file and that the locations are zero-indexed. So, for example, to explain only the first 50 responses in the file, we should set a value of ``"0-49"`` for this option. Defaults to ``None``.
+
+sample_size *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~
+If we want to explain a random sample of the responses in explainable_data, we can specify the size of that random sample here. For example, to explain a random sample of 10 responses, we would set this to 10. Defaults to None.
+
+.. note ::
+
+    Only one of ``sample_range`` or ``sample_size`` must be specified. If neither is specified, explanations will be generated for the entire set of responses in ``explainable_data`` which could be very slow, depending on its size.
+
+show_auto_cohorts *(Optional)*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If this option is set to ``true``, auto cohort bar plots will be displayed. These plots can be useful to detect interaction effects between cohorts and features. If a cohort shows a high feature value, then there may be an interaction between that cohort and the feature. Defaults to ``false``.
```

Comment:
why is this optional? is it very slow?

Reply:
Good question, actually. May be it's because they aren't very helpful if you have a lot of features? @remo-help can you shed some light on why we made this optional? I can't remember now.

Reply:
The reason I set this to optional initially was because this feature can be unstable or provide misleading information if the users do not pass a large dataset for explanation 
I think it was also sometimes a pretty wonky if it could not establish clear cohorts, because the way this works is that it tries to automatically detect possible cohorts (think something like: "rows where feature 24 is > .56 ") and if there are no strong interactions the graphs just look weird

Reply:
it should be unrelated to the total amount of features, since you can still potentially get strong interaction effects between features even if you have a lot, but it might become less likely